packages/backend/src/apps/custom-api/actions/http-request/index.ts

```diff
@@ -61,33 +61,33 @@ const action: IRawAction = {
       throw new Error('The URL you are trying to call is not allowed.')
     }
 
-    const response = await $.http
-      .request({
+    try {
+      const response = await $.http.request({
         url,
         method,
         data,
         // Redirects open up way too many vulns (e.g. someone changes the
         // redirect target to a malicious endpoint), so disable it.
         maxRedirects: 0,
       })
-      .catch((err): never => {
-        const stepErrorSolution =
-          'Check your custom app based on the status code and retry again.'
-        throw generateHttpStepError(
-          err,
-          stepErrorSolution,
-          $.step.position,
-          $.app.name,
-        )
-      })
 
-    let responseData = response.data
+      let responseData = response.data
 
-    if (typeof response.data === 'string') {
-      responseData = response.data.replaceAll('\u0000', '')
-    }
+      if (typeof response.data === 'string') {
+        responseData = response.data.replaceAll('\u0000', '')
+      }
 
-    $.setActionItem({ raw: { data: responseData } })
+      $.setActionItem({ raw: { data: responseData } })
+    } catch (err) {
+      const stepErrorName = `Status code: ${err.response.status} (${err.response.statusText})`
```

Comment:
Can just in-line the variable if it's only used once

Reply:
okay, thanks!